webrtc-stats.html

```diff
@@ -307,15 +277,47 @@ <h3>
       </section>
       <section>
         <h3>
-          Stats objects for deleted monitored objects
+          Lifetime considerations for monitored objects
         </h3>
         <p>
-          When a monitored object is destroyed, a final stats object is produced, carrying the
-          values current at the time of destruction. This object will be returned, with a timestamp
-          reflecting the time of destruction, whenever getStats() is called, as long as the
-          PeerConnection exists. This is important in order to report consistently on short-lived
-          objects and to be able to consistently report totals over the lifetime of a
-          PeerConnection.
+          The object descriptions will say what the lifetime of a <a>monitored object</a> from the
+          perspective of stats is. When a monitored object is "deleted", it no longer appears in stats;
+          until this happens, it will appear. This may or may not correspond to the actual lifetime
+          of an object in an implementation; what matters for this specification is what appears in stats.
+        </p>
+        <p>
+          If a monitored object can only exist
+          in a few instances over the lifetime of a PeerConnection, it may be simplest to consider it
+          "eternal" and never delete it from the set of objects reported on in stats.
+          This type of object will remain visible until the PeerConnection
+          is no longer available; it is also visible in getStats() after pc.close(). This is the default
+          when no lifetime is mentioned in its specification.
+        </p>
+        <p>
+          Objects that might exist in many instances over time should have a defined time at which
+          they are deleted, at which time a <a>statsended</a> event is fired on their behalf. Each
+          event that causes deletions to happen MUST fire only one <a>statsended</a> event, but there are
+          cases where one action causes multiple deletion events; for instance,
+          an ICE restart will fire only one event containing the stats for all the discarded candidates
+          and pairs, but will also cause a later event to be fired when the currently-in-use candidate
+          pair and its candidates are discarded.
+        </p>
+        <p>
+          When a <a>monitored object</a> is deleted, a final stats object is produced, carrying the
+          values current at the time of deletion. This object will be made available, with a
+          timestamp reflecting the time of deletion, using the
```

Comment:
Will it always be the case that the timestamp of the stats object reflect the time of deletion?

For example, RTCInboundRtpStreamStats has "The timestamp reported in the statistics object is the time at which the data was sampled." and RTCRemoteInboundRtpStreamStats has "The timestamp reported in the statistics object is the time at which the corresponding RTCP RR was received."
The spec contradicts itself, either we lose the ability to know when the corresponding RR was received or we lose the ability to know when the monitored object was deleted.

Should we add a timestamp to RTCStatsReport or to the statsended event?
I'm fine with discussing this separately if we want to land this.

Reply:
It's not clear that it makes sense ever to delete remote objects; we don't know if they're gone, or if another RR is imminent some time in the near future. I'd consider this more closely before making changes.

The InboundRtpStreamStats timestamp is consistent with the time of deletion; we sample at the time of deletion.

Reply:
Deleted the subsentence. The start says "carrying the values current at the time of deletion"; the timestamp should reflect the time these stats were gathered (whatever that is).